devlib/target.py

```diff
@@ -153,6 +155,7 @@ def __init__(self,
                  ):
         self.connection_settings = connection_settings or {}
         self.platform = platform or Platform()
+        self.connection_settings['platform'] = self.platform
```

Comment:
The `platform` argument needs to be added in all the other connections (e.g. `ADBConnection`).
Also there should be a check here to ensure that platform is not already set in `connection_settings`.